codegenerator/cli/npm/envio/src/db/Table.res

```diff
@@ -166,6 +178,92 @@ let getUnfilteredCompositeIndicesUnsafe = (table): array<array<string>> => {
   )
 }
 
+type sqlParams<'entity> = {
+  dbSchema: S.t<'entity>,
+  quotedFieldNames: array<string>,
+  quotedNonPrimaryFieldNames: array<string>,
+  arrayFieldTypes: array<string>,
+  hasArrayField: bool,
+}
+
+let toSqlParams = (table: table, ~schema) => {
+  let quotedFieldNames = []
+  let quotedNonPrimaryFieldNames = []
+  let arrayFieldTypes = []
+  let hasArrayField = ref(false)
+
+  let dbSchema: S.t<Js.Dict.t<unknown>> = S.schema(s =>
+    switch schema->S.classify {
+    | Object({items}) =>
+      let dict = Js.Dict.empty()
+      items->Belt.Array.forEach(({location, inlinedLocation, schema}) => {
+        let rec coerceSchema = schema =>
+          switch schema->S.classify {
+          | BigInt => BigInt.schema->S.toUnknown
+          | Option(child)
+          | Null(child) =>
+            S.null(child->coerceSchema)->S.toUnknown
+          | Array(child) => {
+              hasArrayField := true
+              S.array(child->coerceSchema)->S.toUnknown
+            }
+          | JSON(_) => {
+              hasArrayField := true
+              schema
+            }
+          | Bool =>
+            // Workaround for https://github.com/porsager/postgres/issues/471
+            S.union([
+              S.literal("t")->S.to(_ => true),
+              S.literal("f")->S.to(_ => false),
+            ])->S.toUnknown
```

Comment:
Am I understanding correctly that the whole purpose of this function is to transform booleans and flag in the case an array/json type exists? But still to keep the same underlying schema otherwise?

Reply:
Yes, besides the part of `array/json`. `array/json` has nothing to do with it. 

1. We always convert bool values, since before it was done by `sql` call
2. Array/json check needed to decide whether we want to use `insert unnest` or fallback to the old query.